Automattic/simplenote-macos
# Decoupling PopoverWindowController

### Details
- Renames **InterlinkWindowController** > **PopoverWindowController**
- Renames **AutocompleteWindow** > **PopoverWindow**
- Drops **Interlink.storyboard**

cc @eshurakov 
(Thanks in advance!!)

### Test
1. Open any note
2. Type `[` + keyword (Note that **keyword** must be present in any of your note's titles)

- [x] Verify the Interlink Popover shows up
- [x] Verify that you can move the Keyboard Arrows (Up / Down) to cycle thru results
- [x] Verify that typing anything else causes the UI to refresh the results
- [x] Verify that pressing **Return** inserts the Interlink
- [x] Verify that clicking elsewhere (while the popover is visible) dismisses the Interlink UI

### Release
These changes do not require release notes.
